### PR TITLE
feat: adds ts-runtime-checks

### PR DESCRIPTION
**Pull request by mmamedel**
This is to include this little library that impressed me in simplicity and performance. If one is searching for a library that would give same type-safety of Typescript at runtime, this is a winner. There is also a way of extending some of that validation using Expressions.

### CODE REVIEW

**Review by moltar — APPROVED** [review]

**Comment by moltar** [comment]
LGTM

Assigning to @hoeck for an additional check

**Comment by moltar** [comment]
Huh:

```
npm ERR! code ENOTFOUND
npm ERR! syscall getaddrinfo
npm ERR! errno ENOTFOUND
npm ERR! network request to https://packages.bethel.jw.org/artifactory/api/npm/npm-meps/typescript/-/typescript-4.6.2.tgz failed, reason: getaddrinfo ENOTFOUND packages.bethel.jw.org
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network 
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
```

**Comment by moltar** [comment]
Wtf is `packages.bethel.jw.org`?

File: package-lock.json [hunk 1]
```diff
@@ -9629,6 +9625,12 @@
       "integrity": "sha512-pf6Y4xceimqXFp797dxRtlaD42a91UsJ5vSX6AQUaK44nWF8uyFs2eIZbJ4xmk+lFJSfI2vSA7IYhJdelLgkWA==",
       "dev": true
     },
+    "node_modules/ts-runtime-checks": {
+      "version": "0.1.1",
+      "resolved": "https://packages.bethel.jw.org/artifactory/api/npm/npm-meps/ts-runtime-checks/-/ts-runtime-checks-0.1.1.tgz",
```
Comment (moltar) [inline package-lock.json:9630]: This is weird

Comment (mmamedel) [reply package-lock.json:9630]: Sorry, let me fix this

Comment (mmamedel) [reply package-lock.json]: Fixed

**Review by moltar — CHANGES_REQUESTED** [review]
Sorry, missed one more thing. We use strict versions in the package file. This way we can be sure what is being tested. And renovate bot will update the versions when they get released.

File: package.json [hunk 2]
```diff
@@ -60,7 +61,9 @@
     "ts-interface-checker": "1.0.2",
     "ts-json-validator": "0.7.1",
     "ts-node": "10.9.1",
-    "typescript": "4.5.5",
+    "ts-runtime-checks": "^0.1.1",
+    "ttypescript": "^1.5.13",
+    "typescript": "^4.6.2",
```
[the remark this answers is not in the document]
Comment (mmamedel) [reply package.json]: fixed

File: package.json [hunk 3]
```diff
@@ -11,13 +11,14 @@
   "scripts": {
     "lint": "gts check",
     "lint:fix": "gts fix",
-    "start": "ts-node index.ts",
-    "test:build": "npm run compile:spectypes && tsc --noEmit",
-    "test": "npm run compile:spectypes && jest",
+    "start": "ts-node --compiler ttypescript index.ts",
```
Comment (hoeck) [inline package.json:14]: You already added a separate compile step for ts-runtime-checks, to me it looks that the transformer should not be required?

Using the `--compiler` argument would result in all libraries running though the ttypescript compiler. Would be good if we could avoid this.

File: cases/ts-runtime-checks/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  longString: string;
+  boolean: boolean;
+  deeplyNested: {
+    foo: string;
+    num: number;
+    bool: boolean;
+  };
+}
+
+export const assertLoose = (value: Assert<ToBeChecked>) => true;
```
Comment (hoeck) [inline cases/ts-runtime-checks/src/index.ts:17]: Please add a comment to silence the eslint warning.

**Comment by hoeck** [comment]
> Thank you for the testing, it revealed an issue with the library and additional properties on nested objects.

Thx for adding the library. Also good to know that the tests are helping to understand how the different benchmarks work :grin:

File: package.json (hunk 3 above)
Comment (mmamedel) [reply package.json]: Removed.

File: cases/ts-runtime-checks/src/index.ts (hunk 4 above)
Comment (mmamedel) [reply cases/ts-runtime-checks/src/index.ts]: added ;)

**Comment by moltar** [comment]
@hoeck if you are happy with the changes, please merge! 🎉

**Pull request merged**